pkg/engine/planner/logical/dataframe.go

```diff
@@ -4,123 +4,64 @@ import (
 	"github.com/grafana/loki/v3/pkg/engine/planner/schema"
 )
 
-// DataFrame provides an ergonomic interface for building logical query plans.
-// It wraps a logical Plan and provides fluent methods for common operations
-// like projection, filtering, and aggregation. This makes it easier to build
-// complex query plans in a readable and maintainable way.
+// DataFrame provides an ergonomic builder-like interface for constructing a
+// [Plan].
 type DataFrame struct {
```

Comment:
nit: I would prefer a different name than `DataFrame`, since it is not a two-dimensional datastructure representing a table of rows and columns (unlike [Datafusion Dataframe](https://datafusion.apache.org/user-guide/dataframe.html), [Spark DataFrame](https://spark.apache.org/docs/latest/sql-programming-guide.html), or [Pandas DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html)).
In our case, it it really just a builder.